pkg/synchronization/core/scan.go

```diff
@@ -107,6 +110,10 @@ type scanner struct {
 	ignorer *ignorer
 	// ignoreCache is the cache of ignored path behavior.
 	ignoreCache IgnoreCache
+	// ignorerMode allows customizing how the ignore logic is applied.  The
+	// default value is "gitignore" and will work similar to git ignoring.
+	// The value of "dockerignore" will work similar to docker ignoring.
+	ignorerMode string
```

Comment:
This is the right idea, just needs to be converted to a Protocol Buffers enumeration. You can probably copy/paste the `SymbolicLinkMode` implementation. Even the path of threading that through the session configuration should be identical. In addition to varying this in the integration tests, this will also need to be threaded into the endpoint infrastructure to enable accelerated scans, e.g. [here](https://github.com/mutagen-io/mutagen/blob/af15c085b40d410426c3e80d7d978cdef35f2e50/pkg/synchronization/endpoint/local/endpoint.go#L984).